### PR TITLE
Remove code due to be removed in v0.28

### PR DESCRIPTION
**Pull request by albi3ro**
A variety of code is due to be removed for the v0.28 release.  See [the deprecations page](https://docs.pennylane.ai/en/latest/development/deprecations.html) for the full list.

The list is: 

  - `qml.tape.get_active_tape`: Use `qml.QueuingManager.active_context()`
  - `qml.transforms.qcut.remap_tape_wires`: Use `qml.map_wires`
  - `qml.tape.QuantumTape.inv()`: Use `qml.tape.QuantumTape.adjoint()`
  - `qml.tape.stop_recording()`: Use `qml.QueuingManager.stop_recording()`
  - `qml.tape.QuantumTape.stop_recording()`: Use `qml.QueuingManager.stop_recording()`
  - `qml.QueuingContext` is now `qml.QueuingManager`
  - `QueuingManager.safe_update_info` and `AnnotatedQueue.safe_update_info`: Use plain `update_info`

### CODE REVIEW

**Review by timmysilv — COMMENTED** [review]
image
😍

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -13,6 +13,16 @@
 
 <h3>Deprecations</h3>
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply doc/releases/changelog-dev.md:14]: Good idea 👍

File: doc/development/deprecations.rst [hunk 2]
```diff
@@ -60,33 +54,39 @@ Pending deprecations
...
-  - Will only be accessible via ``QueuingManager`` in v0.28
+Completed deprecation cycles
+----------------------------
 
-* ``QueuingManager.safe_update_info`` and ``AnnotateQueue.safe_update_info`` are removed.
+* ``qml.tape.get_active_tape`` is deprecated. Please use ``qml.QueuingManager.active_context()`` instead.
 
   - Still accessible in v0.27
-  -  Will be removed in v0.28
+  - Removed in v0.28
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply doc/development/deprecations.rst:63]: Yep.  These changes will be released with v0.28.

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -36,6 +36,19 @@
...
 
+Deprecations cycles are tracked at [(docs/developement/deprecations.rst)](https://docs.pennylane.ai/en/latest/development/deprecations.html).
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:39]:
```suggestion
Deprecations cycles are tracked at [doc/developement/deprecations.rst](https://docs.pennylane.ai/en/latest/development/deprecations.html).
```

**Review by timmysilv — APPROVED** [review]
Two notes for consistency but otherwise lgtm!
1. Two of the pending deprecations say "Will be removed in v0.28" - are we sure we'll get to them? Otherwise it will clash with the completed deprecations that say "Removed in v0.28". Not sure what I'm asking for here. Stories to track them? Optimistically change the language now? A calendar reminder to ensure it's updated pre-release? Nothing is also acceptable, just making sure it's pointed out.
2. The newly completed deprecations have the first line saying "Still accessible in vX.Y", while the previously existing ones say "Deprecated in vX.Y". Can we make them the same? I'd prefer "Deprecated in".

**Comment by albi3ro** [comment]
For the grouping deprecation, let's make a separate PR for that @Jaybsoni . Since that will be a lot of deletions and clean up.  

I'm unsure whether we should actually remove `qml.ExpvalCost` in this release, because of the `ShotAdaptiveOptimizer`.  I think we should wait to actually remove it when we are capable of distributing shots over a script batch.

Updated the language for "Completed deprecation cycles".

**Pull request merged**